pkg/reloader/vault.go

```diff
@@ -0,0 +1,173 @@
+// Copyright © 2023 Cisco
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reloader
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/bank-vaults/vault-sdk/vault"
+	vaultapi "github.com/hashicorp/vault/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logrusadapter "logur.dev/adapter/logrus"
+)
+
+type VaultConfig struct {
+	Addr          string
+	AuthMethod    string
+	Role          string
+	Path          string
+	Namespace     string
+	SkipVerify    bool
+	TLSSecret     string
+	TLSSecretNS   string
+	ClientTimeout time.Duration
+}
+
+func getVaultConfig() *VaultConfig {
+	var vaultConfig VaultConfig
+
+	vaultConfig.Addr = os.Getenv("VAULT_ADDR")
+	if vaultConfig.Addr == "" {
+		vaultConfig.Addr = "https://vault:8200"
+	}
+
+	vaultConfig.AuthMethod = os.Getenv("VAULT_AUTH_METHOD")
+	if vaultConfig.AuthMethod == "" {
+		vaultConfig.AuthMethod = "jwt"
+	}
+
+	vaultConfig.Role = os.Getenv("VAULT_ROLE")
+
+	vaultConfig.Path = os.Getenv("VAULT_PATH")
+	if vaultConfig.Path == "" {
+		vaultConfig.Path = "kubernetes"
+	}
+
+	vaultConfig.Namespace = os.Getenv("VAULT_NAMESPACE")
+	if vaultConfig.Namespace == "" {
+		vaultConfig.Namespace = "default"
+	}
+
+	vaultConfig.SkipVerify, _ = strconv.ParseBool(os.Getenv("VAULT_SKIP_VERIFY"))
+
+	vaultConfig.TLSSecret = os.Getenv("VAULT_TLS_SECRET")
+
+	vaultConfig.TLSSecretNS = os.Getenv("VAULT_TLS_SECRET_NS")
+	if vaultConfig.TLSSecretNS == "" {
+		vaultConfig.TLSSecretNS = "default"
+	}
+
+	vaultConfig.ClientTimeout, _ = time.ParseDuration(os.Getenv("VAULT_CLIENT_TIMEOUT"))
+	if vaultConfig.ClientTimeout == 0 {
+		vaultConfig.ClientTimeout = 10 * time.Second
+	}
+
+	return &vaultConfig
+}
+
+func (c *Controller) initVaultClient(config *VaultConfig) error {
+	c.logger.Info("Creating Vault client")
+
```

Comment:
I am not sure if we want to recreate client on every call. We can assume that the client is valid if it's not nil. In addition, I think there's no need to fetch config on every call since these values should not change. 
```suggestion
func (c *Controller) initVaultClient() error {
	if c.vaultClient != nil {
		return nil
	}
	
	c.logger.Info("Creating Vault client")
	config := getVaultConfig()
```

Reply:
Thanks for the suggestion!